test/e2e/specs/editor/various/block-switcher.spec.js

```diff
@@ -8,6 +8,179 @@ test.describe( 'Block Switcher', () => {
+			'aria-haspopup',
+			'true'
+		);
+
+		// Verify the correct block transforms appear.
+		await blockSwitcher.click();
+		await expect(
+			page.getByRole( 'menu', { name: 'List' } ).getByRole( 'menuitem' )
+		).toHaveText( [ 'Paragraph', 'Heading', 'Quote', 'Columns', 'Group' ] );
+	} );
+
+	test( 'Should show the expected block transforms on the list block when the quote block is removed', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		// Remove the quote block from the list of registered blocks.
+		await page.waitForFunction( () => {
+			try {
+				window.wp.blocks.unregisterBlockType( 'core/quote' );
+				return true;
+			} catch {
+				return false;
+			}
+		} );
+
+		// Insert a list block.
+		await editor.canvas
+			.getByRole( 'button', { name: 'Add default block' } )
+			.click();
+		await page.keyboard.type( '- List content' );
+		await page.keyboard.press( 'ArrowUp' );
+		await pageUtils.pressKeys( 'alt+F10' );
+
+		const blockSwitcher = page
+			.getByRole( 'toolbar', { name: 'Block tools' } )
+			.getByRole( 'button', { name: 'List' } );
+
+		// Verify the block switcher exists.
+		await expect( blockSwitcher ).toHaveAttribute(
+			'aria-haspopup',
+			'true'
+		);
+
+		// Verify the correct block transforms appear.
+		await blockSwitcher.click();
+		await expect(
+			page.getByRole( 'menu', { name: 'List' } ).getByRole( 'menuitem' )
+		).toHaveText( [ 'Paragraph', 'Heading', 'Columns', 'Group' ] );
+	} );
+
+	test( 'Should not show the block switcher if all the blocks the list block transforms into are removed', async ( {
+		editor,
+		page,
+		pageUtils,
+	} ) => {
+		// Insert a list block.
+		await editor.canvas
+			.getByRole( 'button', { name: 'Add default block' } )
+			.click();
+		await page.keyboard.type( '- List content' );
+
+		// Remove blocks.
+		await page.waitForFunction( () => {
+			try {
+				window.wp.data
+					.dispatch( 'core/blocks' )
+					.removeBlockTypes( [
+						'core/quote',
+						'core/pullquote',
+						'core/paragraph',
+						'core/group',
+						'core/heading',
+						'core/columns',
+					] );
+				return true;
+			} catch {
+				return false;
+			}
+		} );
+
+		await page.keyboard.press( 'ArrowUp' );
+		await pageUtils.pressKeys( 'alt+F10' );
+
+		// Verify the block switcher isn't enabled.
+		await expect(
+			page
+				.getByRole( 'toolbar', { name: 'Block tools' } )
+				.getByRole( 'button', { name: 'List' } )
+		).toBeDisabled();
+	} );
+
+	test( 'Should show Columns block only if selected blocks are between limits (1-6)', async ( {
+		editor,
+		page,
+	} ) => {
+		await editor.canvas
+			.getByRole( 'button', { name: 'Add default block' } )
+			.click();
+		await page.keyboard.type( '- List content' );
+		await page.keyboard.press( 'ArrowUp' );
+		await page.keyboard.press( 'Enter' );
+		await page.keyboard.type( '## I am a header' );
+		await page.keyboard.down( 'Shift' );
+		await page.keyboard.press( 'ArrowUp' );
+		await page.keyboard.up( 'Shift' );
```

Comment:
Note for future me: If this fails again, ensure the block toolbar is displayed by adding `await pageUtils.pressKeys( 'alt+F10' );`.

Currently, I can reproduce the scenario with Playwright, but I can do it manually sometimes.